forc-plugins/forc-tx/src/lib.rs

```diff
@@ -306,29 +306,231 @@ pub struct OutputContractCreated {
     pub state_root: fuel_tx::Bytes32,
 }
 
+/// Errors that can occur while parsing the `Command`.
+#[derive(Debug, Error)]
+pub enum ParseError {
+    #[error("Failed to parse the command")]
+    Command {
+        #[source]
+        err: clap::Error,
+    },
+    #[error("Failed to parse transaction `input`")]
+    Input {
+        #[source]
+        err: clap::Error,
+    },
+    #[error("Failed to parse transaction `output`")]
+    Output {
+        #[source]
+        err: clap::Error,
+    },
+    #[error("Unrecognized argument {arg:?}, expected `input` or `output`")]
+    UnrecognizedArgumentExpectedInputOutput { arg: String, remaining: Vec<String> },
+    #[error("Found argument `input` which isn't valid for a mint transaction")]
+    MintTxHasInput,
+}
+
+/// Errors that can occur during conversion from the CLI transaction
+/// representation to the `fuel-tx` representation.
+#[derive(Debug, Error)]
+pub enum ConvertTxError {
+    #[error("failed to convert create transaction")]
+    Create(#[from] ConvertCreateTxError),
+    #[error("failed to convert script transaction")]
+    Script(#[from] ConvertScriptTxError),
+}
+
+/// Errors that can occur during "create" transaction conversion.
+#[derive(Debug, Error)]
+pub enum ConvertCreateTxError {
+    #[error("failed to open `--storage-slots` from {path:?}")]
+    StorageSlotsOpen {
+        path: PathBuf,
+        #[source]
+        err: std::io::Error,
+    },
+    #[error("failed to deserialize storage slots file")]
+    StorageSlotsDeserialize(#[source] serde_json::Error),
+    #[error("failed to convert an input")]
+    Input(#[from] ConvertInputError),
+}
+
+/// Errors that can occur during "script" transaction conversion.
+#[derive(Debug, Error)]
+pub enum ConvertScriptTxError {
+    #[error("failed to read `--bytecode` from {path:?}")]
+    BytecodeRead {
+        path: PathBuf,
+        #[source]
+        err: std::io::Error,
+    },
+    #[error("failed to read `--data` from {path:?}")]
+    DataRead {
+        path: PathBuf,
+        #[source]
+        err: std::io::Error,
+    },
+    #[error("failed to convert an input")]
+    Input(#[from] ConvertInputError),
+}
+
+/// Errors that can occur during transaction input conversion.
+#[derive(Debug, Error)]
+pub enum ConvertInputError {
+    #[error("failed to read message data from {path:?}")]
+    MessageDataRead {
+        path: PathBuf,
+        #[source]
+        err: std::io::Error,
+    },
+    #[error("failed to read predicate from {path:?}")]
```

Comment:
```suggestion
    #[error("failed to read --predicate from {path:?}")]
```

Should we be consistent with printing the error here? Above, in the `BytecodeRead` error, we are printing the option flag as is, but here, we're not showing `--predicate`. IMO in this case, explicitly showing the option flag works better since it's a long command.